projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/sample/SampleOrigin.java

```diff
@@ -0,0 +1,15 @@
+package life.qbic.projectmanagement.domain.project.sample;
+
+import life.qbic.projectmanagement.domain.project.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.project.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.project.experiment.vocabulary.Specimen;
+
+/**
+ * <b><class short description - 1 Line!></b>
+ *
+ * <p><More detailed description - When to use, what it solves, etc.></p>
+ *
+ * @since <version tag>
+ */
+public record SampleOrigin(Species species, Specimen specimen, Analyte analyte) {
```

Comment:
:+1: very good.

You could check in the canonical constructor that none of these are null.